src/analysis/vector/geometry_checker/qgsgeometryselfintersectioncheck.h

```diff
@@ -45,7 +45,7 @@ class ANALYSIS_EXPORT QgsGeometrySelfIntersectionCheckError : public QgsSingleGe
 class ANALYSIS_EXPORT QgsGeometrySelfIntersectionCheck : public QgsSingleGeometryCheck
 {
   public:
-    explicit QgsGeometrySelfIntersectionCheck( const QgsGeometryCheckContext *context, const QVariantMap &configuration )
+    explicit QgsGeometrySelfIntersectionCheck( const QgsGeometryCheckContext *context, const QVariantMap &configuration = QVariantMap() )
```

Comment:
@m-kuhn I can't compile (with reference in static_cast, or run with pointer in static_cast) if I don't put the default value for the config here. I really don't understand why.

> /Users/denis/opt/qgis/QGIS/src/analysis/vector/geometry_checker/qgsgeometryselfintersectioncheck.h:45:23: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'int' to 'QgsGeometrySelfIntersectionCheck' for 1st argument
/Users/denis/opt/qgis/QGIS/src/analysis/vector/geometry_checker/qgsgeometryselfintersectioncheck.h:48:14: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    explicit QgsGeometrySelfIntersectionCheck( const QgsGeometryCheckContext *context, const QVariantMap &configuration )

full output: https://pastebin.com/sF6UjVRq

Reply:
Interesting.
Maybe it calls actually a constructor `QgsGeometrySelfIntersectionCheck( 0 /*, configuration = QVariantMap() */ );` like this?